### DISCUSSION TITLE
How to handle nested group hover styles?

### DISCUSSION THREAD

**Discussion by hazem3500** [chakra-ui/chakra-ui#4485] [Q&A] [answered]
I have an `Icon` that is inside a `Button` which is nested inside a `ButtonGroup`
The `Icon` color should change when I hover over the `Button` but not when I hover over the `ButtonGroup`

### Example
```jsx
<ButtonGroup>
    <Button rightIcon={<Icon _groupHover={{ color: "red.500" }} />}>
        Click
    </Button>
    <IconButton icon={<Icon _groupHover={{ color: "green.500" }} />} />
</ButtonGroup>
```

The problem is that `ButtonGroup` and `Button` both have a `role="group"` which makes the `Icon` change color when I hover over the whole `ButtonGroup`

![preview](https://i.imgur.com/hz1YSQp.gif)

**CodeSandBox Link:** https://codesandbox.io/s/tender-poitras-oc206?file=/src/index.js

**Comment by aaronkik** [comment]
Hi Hazem,

This is my solution: https://codesandbox.io/s/blazing-grass-y5m4j?file=/src/index.js

I had to wrap the text in the button with the `Text` component to prevent it from inheritting the red color.

**Reply by hazem3500** [reply]
Oh, that's a cool solution, but what can we do if it's a non-inheritable property like `transform`?

**Reply by aaronkik** [reply]
Hmm I'm not too sure, you may need to handle some state that would let you know if you're hovering over the `Button` container then set some css variables to the `Icon` based on the state.

This is a possible solution the only issue is you would need to apply some animation to it if you want it smooth. 
https://codesandbox.io/s/sad-wright-n2g0b?file=/src/index.js

Here's the hook I used for it: https://chakra-ui.com/docs/hooks/use-boolean

**Reply by hazem3500** [reply]
Hmmmm, I guess that will work, would've been better if we didn't need to use state/js for it though 😅

Thanks @AaronK97 for your help XD

**Comment by anubra266** [comment]
There's a way to do this without state. 
First give a classname to both your button and Icon. Then call the sibling identifier styles in the `ButtonGroup`

```html
<ButtonGroup sx={{ '.but:hover ~ .ico' : { color: 'green.500' } }}>
    <Button className="but" rightIcon={<Icon />}>
        Click
    </Button>
    <IconButton icon={<Icon  className="ico" />} />
</ButtonGroup>
```

**Comment by hazem3500** [comment, marked as answer]
What I ended up doing was add to my `Button` component an attribute `data-component-name="Button"`

then used it as a selector

```js
{
  '[data-component-name="Button"]:hover &': {
    color: "green.500"
  }
}
```

Quite pragmatic, I think every component in Chakra-UI should have a `data-component-name` attribute as an escape-hatch

**Reply by veloware** [reply]
Ive been trying all sorts of weird and wonderful things, knowing that this was achievable, but I've been just trying to source a way where the child component doesn't reference the parent component in any way, as in you have the child component referencing 'Button' within it. Aw well, if that's whats needed... Thanks!

**Comment by Sid-Turner-Ellis** [comment]
@hazem3500 fyi you can also do this:

```
<Box role='group'>
    <Box _groupHover={{bgColor: 'blue.500'}}>
        <Component />
    </Box>

    <OtherComponents />
</Box>
```

Docs: https://chakra-ui.com/docs/styled-system/style-props#pseudo

**Reply by Haribhusal** [reply]
It works for me, Thanks
